error-prone-contrib/src/test/java/tech/picnic/errorprone/refasterrules/RefasterRulesTest.java

```diff
@@ -77,12 +83,24 @@ private static Stream<Arguments> validateRuleCollectionTestCases() {
     // method with `@ValueSource(classes = {...})`.
     return RULE_COLLECTIONS.stream()
         .filter(not(AssertJRules.class::equals))
+        .filter(not(WebClientRules.class::equals))
         .map(Arguments::arguments);
   }
 
   @MethodSource("validateRuleCollectionTestCases")
   @ParameterizedTest
   void validateRuleCollection(Class<?> clazz) {
-    RefasterRuleCollection.validate(clazz);
+    if (!JDK_17_PLUS_RULE_COLLECTIONS.contains(clazz)) {
+      RefasterRuleCollection.validate(clazz);
+    }
+  }
+
+  @DisabledForJreRange(max = JRE.JAVA_16 /* Spring targets JDK 17. */)
+  @MethodSource("validateRuleCollectionTestCases")
+  @ParameterizedTest
+  void validateRuleCollectionWithJdk17AndAbove(Class<?> clazz) {
+    if (JDK_17_PLUS_RULE_COLLECTIONS.contains(clazz)) {
+      RefasterRuleCollection.validate(clazz);
+    }
   }
```

Comment:
We can of course instead loop over `JDK_17_PLUS_RULE_COLLECTIONS`, but I wanted to stick with a single source of parameterized test elements. :shrug:

An alternative approach would be to have a single parameterized test with `(Class<?> clazz, JRE minJre)` parameters, but that seems overkill for now. If/when we need to introduce JDK 21-only tests while still supporting JDK 11 we should reconsider.